snuba/migrations/snuba_migrations/events/0005_events_tags_hash_map.py

```diff
@@ -0,0 +1,62 @@
+from typing import Sequence
+
+from snuba.clickhouse.columns import (
+    Array,
+    Column,
+    UInt,
+    Materialized,
+)
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.migrations import migration, operations
+from snuba.datasets.storages.events_common import TAGS_HASH_MAP_COLUMN
+
+
+class Migration(migration.MultiStepMigration):
+    """
+    Adds the tags hash map column defined as Array(Int64) Materialized with
+    arrayMap((k, v) -> cityHash64(
+        concat(replaceRegexpAll(k, '(\\=|\\\\)', '\\\\\\1'), '=', v)
+    ), tags.key, tags.value
+
+    This allows us to quickly find tag key-value pairs since we can
+    add an index on this column.
```

Comment:
Will you be adding the index as a subsequent migration? Or do you want to just add it as an operation here?

Reply:
I actually should add it now. Thanks

Reply:
Actually no, I need to ensure everyone has a recent version of clickhouse first 
```
snuba.clickhouse.errors.ClickhouseError: [36] DB::Exception: You must set the setting `allow_experimental_data_skipping_indices` to 1 before using data skipping indices..
```

So I will add the index separately

Reply:
How will you ensure everyone has a recent version of ClickHouse?

Do you think it's reasonable to assume that anyone with an older version of ClickHouse can just change that setting and rerun it if experimental data skipping indices is not turned on?

Reply:
The clickhouse version is pinned in the onpremise installer, so when you update you update clickhouse as well.
https://github.com/getsentry/onpremise/blob/master/docker-compose.yml#L99

Now as long as we introduce the index in a version of snuba that has a recent clickhouse enough pinned, there should not be any problem.